app/controllers/line_items_controller.rb

```diff
@@ -1,7 +1,10 @@
 class LineItemsController < BaseController
   respond_to :json
 
-  def index
+  # Taken from Spree::Api::BaseController
+  rescue_from ActiveRecord::RecordNotFound, :with => :not_found
```

Comment:
Point taken. As you say, this is a prime example of another convention that should be defined in a lower level controller. Until we make a decision on direction, I am more comfortable implementing something that looks a bit more like a convention than a one-off `before_filter`, but perhaps that is not a very good reason to do it this way in this instance...